tesseract/tesseract_planning/tesseract_time_parameterization/src/iterative_spline_parameterization.cpp

```diff
@@ -340,33 +405,33 @@ bool IterativeSplineParameterization::compute(const std::vector<std::reference_w
     t2[j].accelerations_[num_points - 1] = t2[j].final_acceleration_;
 
     // Set bounds based on inputs
-    t2[j].max_velocity_ = max_velocity[j];
-    t2[j].min_velocity_ = -max_velocity[j];
-    t2[j].max_velocity_ *= velocity_scaling_factor;
-    t2[j].min_velocity_ *= velocity_scaling_factor;
+    t2[j].max_velocity_ = Eigen::VectorXd::Ones(static_cast<Eigen::Index>(num_points)) * max_velocity[j];
+    t2[j].min_velocity_ = Eigen::VectorXd::Ones(static_cast<Eigen::Index>(num_points)) * -max_velocity[j];
+    t2[j].max_velocity_.array() *= velocity_scaling_factor.array();
```

Comment:
Does a VectorXd not have cwiseProduct? I am not sure which performs better.

```
t2[j].max_velocity_ = t2[j].max_velocity_.cwiseProduct(velocity_scaling_factor);
```

Reply:
I found this in the documentation so there is no performance difference. I will leave it up to you which you prefer. 

```
The main difference between the two API is that the one based on cwise* methods returns an expression in the matrix world, while the second one (based on .array()) returns an array expression. Recall that .array() has no cost, it only changes the available API and interpretation of the data.
```

Reply:
I'd probably leave it like this as it seems a bit more concise